README.md

```diff
@@ -279,6 +280,107 @@ module.exports = {
 }
 ```
 
+### Asynchronous code
+
+Asynchronous code can be achieved by using `async` methods:
+
+```js
+// index.js
+
+module.exports = {
+  onPreBuild: async ({ utils }) => {
+    try {
+      await doSomethingAsync()
+    } catch (error) {
+      utils.build.failBuild('Failure message', { error })
+    }
+  },
+}
+```
+
+Any thrown `Error` or rejected `Promise` that is not handled by [`utils.build`](#error-reporting) will be reported as
+bugs.
+
+```js
+// index.js
+
+module.exports = {
+  onPreBuild: async ({ utils }) => {
+    // Any error thrown inside this function will be reported as a bug.
+    await doSomethingAsync()
+  },
+}
+```
+
+Plugins end as soon as their methods end. Therefore you should `await` any asynchronous operation. The following
+examples show invalid code and the way to fix it:
+
+```js
+// index.js
+const { promisify } = require('util')
+
+module.exports = {
+  // This callback will not be awaited
+  onPreBuild: ({ utils }) => {
+    doSomethingAsync((error, response) => {
+      console.log(response)
+    })
+  },
+
+  // This callback will be awaited
+  onPostBuild: async ({ utils }) => {
+    const response = await promisify(doSomethingAsync)()
+    console.log(response)
+  },
+}
+```
```

Comment:
Is this whole sample the invalid example? If so, I recommend making this super clear with an all-caps comment so people don't just copy-paste it without reading the intro text. 😛

Reply:
Fixed in c55304df136e6a68b3d836bb3b4fdfd3877c3794